authentication/auth.go

```diff
@@ -101,96 +100,138 @@ func loadSigningKey() (*PrivateKeyResult, *KeyLoadError) {
+		RuRef:                 postValues.Get("ru_ref"),
+		RuName:                postValues.Get("ru_name"),
+		RefPStartDate:         postValues.Get("ref_p_start_date"),
+		RefPEndDate:           postValues.Get("ref_p_end_date"),
+		ReturnBy:              postValues.Get("return_by"),
+		TradAs:                postValues.Get("trad_as"),
+		EmploymentDate:        postValues.Get("employment_date"),
+		RegionCode:            postValues.Get("region_code"),
+		LanguageCode:          postValues.Get("language_code"),
+		TxID:                  uuid.NewV4().String(),
+		Roles:                 postValues.Get("roles"),
+		VariantFlags: variantFlags{
+			SexualIdentity: postValues.Get("sexual_identity") == "true",
+		},
 	}
 
-	// Add optional values
-	if launcherSchema.URL != "" {
-		claims.SurveyURL = launcherSchema.URL
-	}
+	return postClaims
+}
 
-	if postValues.Get("ref_p_end_date") != "" {
-		claims.RefPEndDate = postValues.Get("ref_p_end_date")
-	}
+// GenerateJwtClaims creates a jwtClaim needed to generate a token
+func GenerateJwtClaims() (jwtClaims jwt.Claims) {
+	issued := time.Now()
+	expires := issued.Add(time.Minute * 10) // TODO: Support custom exp: r.PostForm.Get("exp")
 
-	if postValues.Get("trad_as") != "" {
-		claims.TradAs = postValues.Get("trad_as")
+	jwtClaims = jwt.Claims{
+		IssuedAt: jwt.NewNumericDate(issued),
+		Expiry:   jwt.NewNumericDate(expires),
+		ID:       uuid.NewV4().String(),
 	}
 
-	if postValues.Get("employment_date") != "" {
-		claims.EmploymentDate = postValues.Get("employment_date")
-	}
+	return jwtClaims
+}
 
-	if postValues.Get("region_code") != "" {
-		claims.RegionCode = postValues.Get("region_code")
-	}
+func launcherSchemaFromPost(postValues url.Values) (launcherSchema surveys.LauncherSchema) {
+	schema := postValues.Get("schema")
+	launcherSchema = surveys.FindSurveyByName(schema)
 
-	if postValues.Get("language_code") != "" {
-		claims.LanguageCode = postValues.Get("language_code")
-	}
+	return launcherSchema
+}
 
-	if postValues.Get("roles") != "" {
-		claims.Roles = postValues.Get("roles")
-	}
+func launcherSchemaFromURL(url string) (launcherSchema surveys.LauncherSchema) {
+	resp, _ := http.Get(url)
 
-	if postValues.Get("tx_id") == "true" {
-		claims.TxID = uuid.NewV4().String()
+	responseBody, _ := ioutil.ReadAll(resp.Body)
+
+	var schema Schema
+	if err := json.Unmarshal(responseBody, &schema); err != nil {
+		panic(err)
 	}
 
-	claims.VariantFlags = &variantFlags{
-		SexualIdentity: postValues.Get("sexual_identity") == "true",
+	launcherSchema = surveys.LauncherSchema{
+		EqID:     schema.EqID,
+		FormType: schema.FormType,
+		URL:      url,
 	}
 
-	return claims
+	return launcherSchema
+}
+
+func addToClaims(claims EqClaims, LauncherSchema surveys.LauncherSchema) (finalClaims EqClaims) {
+	finalClaims = claims
+
+	finalClaims.EqID = LauncherSchema.EqID
+	finalClaims.FormType = LauncherSchema.FormType
+	finalClaims.SurveyURL = LauncherSchema.URL
+
+	return finalClaims
```

Comment:
- We should rename the method `addSchemaToClaims`
- I think we can remove the finalClaims return value and pass the claims by reference rather than value, so we'd have:
  ```
  func addSchemaToClaims(claims *EqClaims, LauncherSchema surveys.LauncherSchema) {
  ```
  Which would be called as `addSchemaToClaims(&claims, launcherSchema)`